bin/processGsAndCopyRawDataToPrm.sh

```diff
@@ -0,0 +1,636 @@
+#!/bin/bash
+
+#
+##
+### Environment and Bash sanity.
+##
+#
+if [[ "${BASH_VERSINFO}" -lt 4 || "${BASH_VERSINFO[0]}" -lt 4 ]]
+then
+	echo "Sorry, you need at least bash 4.x to use ${0}." >&2
+	exit 1
+fi
+
+set -e # Exit if any subcommand or pipeline returns a non-zero exit status.
+set -u # Raise exception if variable is unbound. Combined with set -e will halt execution when an unbound variable is encountered.
+set -o pipefail # Fail when any command in series of piped commands failed as opposed to only when the last command failed.
+
+umask 0027
+
+# Env vars.
+export TMPDIR="${TMPDIR:-/tmp}" # Default to /tmp if $TMPDIR was not defined.
+SCRIPT_NAME="$(basename ${0})"
+SCRIPT_NAME="${SCRIPT_NAME%.*sh}"
+INSTALLATION_DIR="$(cd -P "$(dirname "${0}")/.." && pwd)"
+LIB_DIR="${INSTALLATION_DIR}/lib"
+CFG_DIR="${INSTALLATION_DIR}/etc"
+HOSTNAME_SHORT="$(hostname -s)"
+ROLE_USER="$(whoami)"
+REAL_USER="$(logname 2>/dev/null || echo 'no login name')"
+
+
+#
+##
+### Functions.
+##
+#
+
+if [[ -f "${LIB_DIR}/sharedFunctions.bash" && -r "${LIB_DIR}/sharedFunctions.bash" ]]
+then
+	source "${LIB_DIR}/sharedFunctions.bash"
+else
+	printf '%s\n' "FATAL: cannot find or cannot access sharedFunctions.bash"
+	trap - EXIT
+	exit 1
+fi
+
+
+function sanityChecking() {
+	
+	local _run="${1}"
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Processing ${_run}..."
+	
+	#
+	local _controlFileBase="${TMP_ROOT_DIR}/logs/${_run}/${_run}.${SCRIPT_NAME}"
+	local _logFile="${_controlFileBase}.log"
+	
+	#
+	# Determine if samplesheet merging is possible for this run, which is the case when
+	#  1. The data transfer sequence run has finished successfully, and 
+	#  2. corresponding samplesheets are present in samplesheets dir.
+	#  3. samplesheets are ok.
+	#  3. checksums are ok.
+	
+	#
+	# Check if transfer of raw data has finished.
+	#
+	if  [[ -e "${TMP_ROOT_DIR}/${_run}/${_run}.finished" ]]
+	then
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${TMP_ROOT_DIR}/${_run}/${_run}.finished present."
+	else
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${TMP_ROOT_DIR}/${_run}/${_run}.finished absent."
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Skipping ${_run}."
+		return
+	fi
+	
+	#
+	# Check if processGsDataForPrm script was finished before for this run.
+	#
+	if [[ -e "${_controlFileBase}.finished" ]]
+	then
+		#
+		# and check if *_processGsDataForPrm.finished is newer than *.processGsDataForPrm,
```

Comment:
The comments suggest, that there is also a check using timestamps for whether the data was updated, but I don't see that in the code...